public/app/plugins/panel/geomap/layers/data/markersLayer.tsx

```diff
@@ -122,9 +122,26 @@ export const markersLayer: MapLayerRegistryItem<MarkersConfig> = {
 
           // Track if we find any line strings during feature processing
           let hasLineString = false;
+          
+          // Track coordinates to avoid rendering duplicate markers at the same location
+          const processedCoordinates = new Set<string>();
 
           source.forEachFeature((feature) => {
-            const isLineString = feature.getGeometry()?.getType() === 'LineString';
+            const geometry = feature.getGeometry();
+            const isLineString = geometry?.getType() === 'LineString';
+            
+            // For point geometries, check if we've already processed this coordinate
+            if (geometry?.getType() === 'Point') {
+              const coordinates = geometry.getCoordinates();
+              const coordKey = `${coordinates[0]},${coordinates[1]}`;
```

Comment:
A limitation on this approach are for markers with dimension driven color or size. Might need to check for those too to define uniqueness. But, if we do that, it opacities will still stack. Open to suggestions here.

Reply:
hmm, thic could get quite expensive for webgl-quantity of markers? maybe we can take a different approach here?

instead of drawing the additional 0.05 bounding circle, is it possible to clamp the min opacity of the marker itself to 0.05 until the OL issue is fixed? that way we have the same innacurate alpha composition, but without the circle artifact.

Reply:
Great point about performance! I did some testing and this block of code adds the following amount of processing time during initialization:
100k points: ~140ms
10k points: ~15ms
1k points: ~1ms

Considering the current limitation of hit detection when using webGL, I think it is worth the performance hit for now until we find an alternative approach. I'm going to see how much more performance suffers when taking into account color or size as well to define uniqueness.

Reply:
Taking into account color, text, size, and rotation impacts our performance a bit:
100k points: ~180ms
10k points: ~20ms

Reply:
Here's an example of multiple points with the same coordinates that have different sizes and colors.

<img width="426" alt="image" src="https://github.com/user-attachments/assets/f4e8338f-8f5a-4451-90e4-defc3ffe11f1" />

I figure if the markers have distinct dims, they should stack in that case.